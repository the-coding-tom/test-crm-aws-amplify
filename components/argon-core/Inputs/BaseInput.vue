<template>
  <div class="form-group">
    <slot name="label">
      <label
        v-if="label"
        :class="labelClasses">
        {{ label }}
      </label>
    </slot>
    <div
      :class="[
        {'input-group': hasIcon},
        {'focused': focused},
        {'input-group-alternative': alternative},
        {'has-label': label || $slots.label},
        inputGroupClasses
    ]">
      <div
        v-if="prependIcon || $slots.prepend"
        class="input-group-prepend">
        <span class="input-group-text">
          <slot name="prepend">
            <i :class="prependIcon"/>
          </slot>
        </span>
      </div>
      <slot v-bind="slotData">
        <input
          :value="value"
          :type="type"
          v-bind="$attrs"
          :valid="!error"
          :required="required"
          :class="[{'is-valid': valid === true}, {'is-invalid': error}, inputClasses]"
          class="form-control"
          v-on="listeners">
      </slot>
      <div
        v-if="appendIcon || $slots.append"
        class="input-group-append">
        <span class="input-group-text">
          <slot name="append">
            <i :class="appendIcon"/>
          </slot>
        </span>
      </div>
      <slot name="infoBlock"/>
      <slot name="error">
        <div
          v-if="error"
          class="invalid-feedback"
          style="display: block;">
          {{ error }}
        </div>
      </slot>
      <slot name="success">
        <div
          v-if="!error && valid"
          class="valid-feedback">
          {{ successMessage }}
        </div>
      </slot>
    </div>
  </div>
</template>
<script>
export default {
  name: 'BaseInput',
  inheritAttrs: false,
  props: {
    required: {
      type: Boolean,
      description: 'Whether input is required (adds an asterix *)',
      default: true
    },
    group: {
      type: Boolean,
      description:
        'Whether input is an input group (manual override in special cases)',
      default: null
    },
    valid: {
      type: Boolean,
      description: 'Whether is valid',
      default: undefined
    },
    alternative: {
      type: Boolean,
      description: 'Whether input is of alternative layout',
      default: false
    },
    label: {
      type: String,
      description: 'Input label (text before input)',
      default: null
    },
    error: {
      type: String,
      description: 'Input error (below input)',
      default: null
    },
    successMessage: {
      type: String,
      description: 'Input success message',
      default: 'Looks good!'
    },
    labelClasses: {
      type: String,
      description: 'Input label css classes',
      default: 'form-control-label'
    },
    inputClasses: {
      type: String,
      description: 'Input css classes',
      default: null
    },
    inputGroupClasses: {
      type: String,
      description: 'Input group css classes',
      default: null
    },
    value: {
      type: [String, Number],
      description: 'Input value',
      default: null
    },
    type: {
      type: String,
      description: 'Input type',
      default: 'text'
    },
    appendIcon: {
      type: String,
      description: 'Append icon (right)',
      default: null
    },
    prependIcon: {
      type: String,
      description: 'Prepend icon (left)',
      default: null
    }
  },
  data() {
    return {
      focused: false
    }
  },
  computed: {
    listeners() {
      return {
        ...this.$listeners,
        input: this.updateValue,
        focus: this.onFocus,
        blur: this.onBlur
      }
    },
    slotData() {
      return {
        focused: this.focused,
        error: this.error,
        ...this.listeners
      }
    },
    hasIcon() {
      const { append, prepend } = this.$slots
      return (
        append !== undefined ||
        prepend !== undefined ||
        this.appendIcon !== undefined ||
        this.prependIcon !== undefined ||
        this.group
      )
    }
  },
  methods: {
    updateValue(evt) {
      let value = evt.target.value
      this.$emit('input', value)
    },
    onFocus(evt) {
      this.focused = true
      this.$emit('focus', evt)
    },
    onBlur(evt) {
      this.focused = false
      this.$emit('blur', evt)
    }
  }
}
</script>
<style>
</style>
