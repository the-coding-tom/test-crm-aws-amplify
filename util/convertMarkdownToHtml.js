export const convertMarkdownToHtml = text => {
  var showdown = require('showdown')
  var converter = new showdown.Converter()
  const eventDescription = text.replace(/([ ]*\n)/g, `<br/>\n`)
  const finalDescription = eventDescription.replace(
    /(<br\/>\n<br\/>\n<br\/>)/g,
    `<br/><br/>`
  )
  return converter.makeHtml(finalDescription)
}
