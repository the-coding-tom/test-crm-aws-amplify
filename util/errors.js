export const displayError = function(e) {
  if (e.response) {
    processError(e.response.data.message)

    if (e.response.data.errors) {
      _.each(e.response.data.errors, (o, i) => {
        return _.each(o, (v, k) => processError(v))
      })
    }

    return
  }

  processError(e.message)
}

const processError = function(message) {
  console.log('------------------------------------')
  console.log(message)
  console.log('------------------------------------')
}
