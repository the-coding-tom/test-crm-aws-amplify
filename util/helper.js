const randomColor = () => {
  let letters = 'ABCDEF0123456789'
  let len = 6
  let result = '#'
  for (let i = 0; i < len; i++) {
    result += letters[parseInt(Math.random() * letters.length)]
  }
  return result
}

const notify = {
  error: {
    title: 'Error',
    variant: 'danger',
    solid: true
  },
  sucess: {
    title: 'Success',
    variant: 'success',
    solid: true
  }
}

const parseRoomdate = (data, vm) => {
  const roomAvaila = data.split(',')
  const roomAvailability = []
  roomAvaila.forEach(data => {
    let weekday = []
    let timeof = []
    let lookup = {
      su: 0,
      mo: 1,
      tu: 2,
      we: 3,
      th: 4,
      fr: 5,
      sa: 6
    }
    const timeData = data.split(' ')
    if (timeData.length === 2) {
      weekday.push(timeData[0])
      timeof.push(timeData[1])
    }
    if (timeData.length === 3) {
      weekday.push(timeData[1])
      timeof.push(timeData[2])
    }
    const splitWeekday = weekday[0].split('-')
    const splitTimeOf = timeof[0].split('-')
    const AStruc = {
      from: '',
      to: '',
      weekdays: []
    }
    splitWeekday.forEach(week => {
      AStruc.weekdays.push(lookup[week])
    })
    AStruc.from = vm.$moment(splitTimeOf[0], 'HH').format('hh:mm')
    AStruc.to = vm.$moment(splitTimeOf[1], 'HH').format('hh:mm')
    roomAvailability.push(AStruc)
  })
  return roomAvailability
}

const undoParse = (data, vm) => {
  let str = ''
  const lookup = {
    0: 'su',
    1: 'mo',
    2: 'tu',
    3: 'we',
    4: 'th',
    5: 'fr',
    6: 'sa'
  }
  data.forEach(item => {
    let available = ''
    item.weekdays.forEach(day => {
      available += `${lookup[day]}-`
    })
    available = available.slice(0, -1)
    let weekdayfrom = item.from.charAt(0) + item.from.charAt(1)
    let weekdayto = item.to.charAt(0) + item.to.charAt(1)
    available += ` ${weekdayfrom}-${weekdayto}`
    str += `${available},`
  })
  str = str.slice(0, -1)
  return str
}

export default { randomColor, parseRoomdate, notify, parseRoomdate, undoParse }
