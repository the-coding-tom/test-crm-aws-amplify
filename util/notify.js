const notify = {
  error: {
    title: 'Error',
    variant: 'danger',
    solid: true
  },
  sucess: {
    title: 'Success',
    variant: 'success',
    solid: true
  }
}

export default notify
