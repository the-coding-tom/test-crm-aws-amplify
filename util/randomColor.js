const randomColor = () => {
  let letters = 'ABCDEF0123456789'
  let len = 6
  let result = '#'
  for (let i = 0; i < len; i++) {
    result += letters[parseInt(Math.random() * letters.length)]
  }
  return result
}

export default randomColor
