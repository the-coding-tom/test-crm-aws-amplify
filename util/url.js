export const getQueryParams = url => {
  const urlParam = require('url')

  const parsedUrl = urlParam.parse(url)

  return parsedUrl.search
}
