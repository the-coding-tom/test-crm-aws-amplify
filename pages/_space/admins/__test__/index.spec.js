import AddPage from '@/pages/_space/admins/add.vue'
import IndexPage from '@/pages/_space/admins/index.vue'
import EditPage from '@/pages/_space/admins/_id/edit.vue'
import BaseInput from '~/components/argon-core/Inputs/BaseInput.vue'
import BootstrapVue from 'bootstrap-vue'
import BasePagination from '~/components/argon-core/BasePagination'

import VueRouter from 'vue-router'
import cloneDeep from 'lodash/cloneDeep'
import flushPromises from 'flush-promises'
import { createWrapper } from '@/test.util.js'

import allAdmins from './resources/all-admins.json'
import emptyAdmins from './resources/empty-admins.json'

describe('Page: /:space/admins/ | Admins Index Page', () => {
  let NuxtStore, store, wrapper, $admin, router

  beforeAll(async () => {
    NuxtStore = await import(`${process.env.buildDir}/store.js`)
    const { createRouter } = await import(`${process.env.buildDir}/router.js`)

    createRouter()

    $admin = {
      getAllAdmins: jest
        .fn()
        .mockResolvedValueOnce(cloneDeep(emptyAdmins))
        .mockResolvedValue(cloneDeep(allAdmins)),
      deleteAnAdmin: jest
        .fn()
        .mockResolvedValue({ message: 'deleted successfully' })
    }
  })

  beforeEach(async () => {
    store = await NuxtStore.createStore()

    router = new VueRouter({
      routes: [
        {
          path: '_space/admins/add',
          name: 'space-admins-add',
          component: AddPage
        },
        {
          path: '_space/admins/:id/edit',
          name: 'space-admins-id-edit',
          component: EditPage
        }
      ]
    })

    wrapper = createWrapper(
      IndexPage,
      store,
      router,
      {
        loading: false
      },
      {},
      vueInstance => {
        vueInstance.use(BootstrapVue)
        vueInstance.component(BasePagination.name, BasePagination)
        vueInstance.component(BaseInput.name, BaseInput)
        vueInstance.prototype.$admin = $admin
      }
    )

    await wrapper.vm.$options.asyncData({ store, $admin })
  })

  it('renders properly', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('shows all admins', async () => {
    const wrapArr = wrapper.findAll('.cell h4')

    expect(wrapArr.length).toBe(2)
    expect(wrapArr.at(0).text()).toBe(allAdmins.data[0].name)
    expect(wrapArr.at(1).text()).toBe(allAdmins.data[1].name)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('removes admin on click', async () => {
    wrapper.find('button.btn.text-danger.btn-transparent').trigger('click')
    await flushPromises()

    const wrapArr = wrapper.findAll('.cell h4')

    expect(wrapArr.length).toBe(1)
    expect(wrapArr.at(0).text()).toBe(allAdmins.data[1].name)
  })

  it('renders add admin route', async () => {
    wrapper.find('a.btn.text-white.btn-primary.rounded-0').trigger('click')
    await flushPromises()

    expect(wrapper.vm.$route.name).toBe('space-admins-add')
    expect(wrapper.vm.$route.fullPath).toBe('_space/admins/add')
  })

  it('renders edit admin route', async () => {
    wrapper.find('a.btn.text-primary.btn-transparent').trigger('click')
    await flushPromises()

    expect(wrapper.vm.$route.name).toBe('space-admins-id-edit')
    expect(wrapper.vm.$route.fullPath).toBe(
      `_space/admins/${allAdmins.data[1].id}/edit`
    )
  })
})
