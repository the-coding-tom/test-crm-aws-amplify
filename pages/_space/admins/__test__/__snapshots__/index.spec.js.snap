// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: /:space/admins/ | Admins Index Page renders properly 1`] = `
<div>
  <div class="header pb-6">
    <div class="container-fluid">
      <div class="header-body">
        <div class="d-flex justify-content-between align-items-center py-4">
          <h1 class="page-title"><span>Adminstrators</span> <b>•</b> Current Admins</h1> <a href="#_space/admins/add" class="btn text-white btn-primary rounded-0" target="_self">Add Adminstrator</a>
        </div>
      </div>
    </div>
  </div>
  <div class="container-fluid mt--6">
    <div class="card-deck flex-column flex-xl-row">
      <div class="card">
        <div class="el-table table-responsive align-items-center table-flush table-striped el-table--fit el-table--scrollable-x el-table--enable-row-hover">
          <div class="hidden-columns">
            <div></div>
            <div></div>
            <div></div>
            <div></div>
          </div>
          <div class="el-table__header-wrapper">
            <table cellspacing="0" cellpadding="0" border="0" class="el-table__header" style="width: 810px;">
              <colgroup>
                <col name="el-table_1_column_1" width="310">
                <col name="el-table_1_column_2" width="140">
                <col name="el-table_1_column_3" width="180">
                <col name="el-table_1_column_4" width="180">
              </colgroup>
              <thead class="">
                <tr class="thead-light">
                  <th colspan="1" rowspan="1" class="el-table_1_column_1     is-leaf is-sortable">
                    <div class="cell">Name<span class="caret-wrapper"><i class="sort-caret ascending"></i><i class="sort-caret descending"></i></span></div>
                  </th>
                  <th colspan="1" rowspan="1" class="el-table_1_column_2     is-leaf is-sortable">
                    <div class="cell">Email<span class="caret-wrapper"><i class="sort-caret ascending"></i><i class="sort-caret descending"></i></span></div>
                  </th>
                  <th colspan="1" rowspan="1" class="el-table_1_column_3  is-center   is-leaf">
                    <div class="cell"></div>
                  </th>
                  <th colspan="1" rowspan="1" class="el-table_1_column_4  is-center   is-leaf">
                    <div class="cell"></div>
                  </th>
                </tr>
              </thead>
            </table>
          </div>
          <div class="el-table__body-wrapper is-scrolling-left">
            <table cellspacing="0" cellpadding="0" border="0" class="el-table__body" style="width: 810px;">
              <colgroup>
                <col name="el-table_1_column_1" width="310">
                <col name="el-table_1_column_2" width="140">
                <col name="el-table_1_column_3" width="180">
                <col name="el-table_1_column_4" width="180">
              </colgroup>
              <tbody>
                <!---->
              </tbody>
            </table>
            <div class="el-table__empty-block" style="height: 100%; width: 810px;"><span class="el-table__empty-text">暂无数据</span></div>
            <!---->
          </div>
          <!---->
          <!---->
          <!---->
          <!---->
          <div class="el-table__column-resize-proxy" style="display: none;"></div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <ul class="pagination justify-content-center">
      <li class="page-item prev-page disabled"><a aria-label="Previous" class="page-link"><span aria-hidden="true"><i aria-hidden="true" class="fa fa-angle-left"></i></span></a></li>
      <li class="page-item active"><a class="page-link">1</a></li>
      <li class="page-item next-page disabled"><a aria-label="Next" class="page-link"><span aria-hidden="true"><i aria-hidden="true" class="fa fa-angle-right"></i></span></a></li>
    </ul>
  </div>
</div>
`;

exports[`Page: /:space/admins/ | Admins Index Page shows all admins 1`] = `
<div>
  <div class="header pb-6">
    <div class="container-fluid">
      <div class="header-body">
        <div class="d-flex justify-content-between align-items-center py-4">
          <h1 class="page-title"><span>Adminstrators</span> <b>•</b> Current Admins</h1> <a href="#_space/admins/add" class="btn text-white btn-primary rounded-0" target="_self">Add Adminstrator</a>
        </div>
      </div>
    </div>
  </div>
  <div class="container-fluid mt--6">
    <div class="card-deck flex-column flex-xl-row">
      <div class="card">
        <div class="el-table table-responsive align-items-center table-flush table-striped el-table--fit el-table--scrollable-x el-table--enable-row-hover el-table--enable-row-transition">
          <div class="hidden-columns">
            <div></div>
            <div></div>
            <div></div>
            <div></div>
          </div>
          <div class="el-table__header-wrapper">
            <table cellspacing="0" cellpadding="0" border="0" class="el-table__header" style="width: 810px;">
              <colgroup>
                <col name="el-table_2_column_5" width="310">
                <col name="el-table_2_column_6" width="140">
                <col name="el-table_2_column_7" width="180">
                <col name="el-table_2_column_8" width="180">
              </colgroup>
              <thead class="">
                <tr class="thead-light">
                  <th colspan="1" rowspan="1" class="el-table_2_column_5     is-leaf is-sortable">
                    <div class="cell">Name<span class="caret-wrapper"><i class="sort-caret ascending"></i><i class="sort-caret descending"></i></span></div>
                  </th>
                  <th colspan="1" rowspan="1" class="el-table_2_column_6     is-leaf is-sortable">
                    <div class="cell">Email<span class="caret-wrapper"><i class="sort-caret ascending"></i><i class="sort-caret descending"></i></span></div>
                  </th>
                  <th colspan="1" rowspan="1" class="el-table_2_column_7  is-center   is-leaf">
                    <div class="cell"></div>
                  </th>
                  <th colspan="1" rowspan="1" class="el-table_2_column_8  is-center   is-leaf">
                    <div class="cell"></div>
                  </th>
                </tr>
              </thead>
            </table>
          </div>
          <div class="el-table__body-wrapper is-scrolling-left">
            <table cellspacing="0" cellpadding="0" border="0" class="el-table__body" style="width: 810px;">
              <colgroup>
                <col name="el-table_2_column_5" width="310">
                <col name="el-table_2_column_6" width="140">
                <col name="el-table_2_column_7" width="180">
                <col name="el-table_2_column_8" width="180">
              </colgroup>
              <tbody>
                <tr class="el-table__row">
                  <td rowspan="1" colspan="1" class="el-table_2_column_5  ">
                    <div class="cell">
                      <h4>Leslie Akorful</h4>
                    </div>
                  </td>
                  <td rowspan="1" colspan="1" class="el-table_2_column_6  ">
                    <div class="cell"><span>anotheradmin@shack15.com</span></div>
                  </td>
                  <td rowspan="1" colspan="1" class="el-table_2_column_7 is-center ">
                    <div class="cell"><a href="#_space/admins/9c8e28a5-00a9-478b-88c6-ded7180b6653/edit" class="btn text-primary btn-transparent" target="_self"><i class="fa fa-pen"></i> Edit
                      </a></div>
                  </td>
                  <td rowspan="1" colspan="1" class="el-table_2_column_8 is-center ">
                    <div class="cell"><button type="submit" class="btn text-danger btn-transparent"><i class="fa fa-times"></i> Remove
                      </button></div>
                  </td>
                </tr>
                <tr class="el-table__row">
                  <td rowspan="1" colspan="1" class="el-table_2_column_5  ">
                    <div class="cell">
                      <h4>Admin</h4>
                    </div>
                  </td>
                  <td rowspan="1" colspan="1" class="el-table_2_column_6  ">
                    <div class="cell"><span>admin@shack15.com</span></div>
                  </td>
                  <td rowspan="1" colspan="1" class="el-table_2_column_7 is-center ">
                    <div class="cell"><a href="#_space/admins/5ffaab91-d886-4fb6-a080-5b10eb94ca4b/edit" class="btn text-primary btn-transparent" target="_self"><i class="fa fa-pen"></i> Edit
                      </a></div>
                  </td>
                  <td rowspan="1" colspan="1" class="el-table_2_column_8 is-center ">
                    <div class="cell"><button type="submit" class="btn text-danger btn-transparent"><i class="fa fa-times"></i> Remove
                      </button></div>
                  </td>
                </tr>
                <!---->
              </tbody>
            </table>
            <!---->
            <!---->
          </div>
          <!---->
          <!---->
          <!---->
          <!---->
          <div class="el-table__column-resize-proxy" style="display: none;"></div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <ul class="pagination justify-content-center">
      <li class="page-item prev-page disabled"><a aria-label="Previous" class="page-link"><span aria-hidden="true"><i aria-hidden="true" class="fa fa-angle-left"></i></span></a></li>
      <li class="page-item active"><a class="page-link">1</a></li>
      <li class="page-item next-page disabled"><a aria-label="Next" class="page-link"><span aria-hidden="true"><i aria-hidden="true" class="fa fa-angle-right"></i></span></a></li>
    </ul>
  </div>
</div>
`;
